### PR TITLE
107736/assign project

### PR DESCRIPTION
**Pull request by jordan-beard**

### CODE REVIEW

File: Dfe.Academies.Academisation.Service/Mappers/Legacy/ProjectAggregate/LegacyProjectServiceModelMapper.cs [hunk 1]
```diff
@@ -109,6 +109,10 @@ internal static LegacyProjectServiceModel MapToServiceModel(this IProject projec
 			KeyStage2PerformanceAdditionalInformation = project.Details.KeyStage2PerformanceAdditionalInformation,
 			KeyStage4PerformanceAdditionalInformation = project.Details.KeyStage4PerformanceAdditionalInformation,
 			KeyStage5PerformanceAdditionalInformation = project.Details.KeyStage5PerformanceAdditionalInformation,
+			
+			AssignedUser = project.Details.AssignedUser?.Id == null
+				? null
+				: new User(project.Details.AssignedUser!.Id, project.Details.AssignedUser.FullName, project.Details.AssignedUser.EmailAddress)
```
Comment (dneed-nimble) [inline Dfe.Academies.Academisation.Service/Mappers/Legacy/ProjectAggregate/LegacyProjectServiceModelMapper.cs:115]: Why !.Id

File: Dfe.Academies.Academisation.Domain/ProjectAggregate/Project.cs [hunk 2]
```diff
@@ -182,7 +182,10 @@ public CommandResult Update(ProjectDetails detailsToUpdate)
 			// key stage performance tables
 			KeyStage2PerformanceAdditionalInformation = detailsToUpdate.KeyStage2PerformanceAdditionalInformation,
 			KeyStage4PerformanceAdditionalInformation = detailsToUpdate.KeyStage4PerformanceAdditionalInformation,
-			KeyStage5PerformanceAdditionalInformation = detailsToUpdate.KeyStage5PerformanceAdditionalInformation,				
+			KeyStage5PerformanceAdditionalInformation = detailsToUpdate.KeyStage5PerformanceAdditionalInformation,
+			
+			// assigned users
+			AssignedUser = new User(detailsToUpdate.AssignedUser!.Id, detailsToUpdate.AssignedUser.FullName, detailsToUpdate.AssignedUser.EmailAddress)
```
[the remark this answers is not in the document]
Comment (jordan-beard) [reply Dfe.Academies.Academisation.Domain/ProjectAggregate/Project.cs:188]: It's to get rid of the warning in this project.
Its telling the compiler that this Id will always have a value.
Examples of those warnings can be found here - 
image

File: Dfe.Academies.Academisation.Service/Mappers/Legacy/ProjectAggregate/LegacyProjectDetailsMapper.cs [hunk 3]
```diff
@@ -108,7 +108,17 @@ internal static ProjectDetails MapNonEmptyFields(this LegacyProjectServiceModel
 			// key stage performance tables
 			KeyStage2PerformanceAdditionalInformation = detailsToUpdate.KeyStage2PerformanceAdditionalInformation ?? existingProject.Details.KeyStage2PerformanceAdditionalInformation,
 			KeyStage4PerformanceAdditionalInformation = detailsToUpdate.KeyStage4PerformanceAdditionalInformation ?? existingProject.Details.KeyStage4PerformanceAdditionalInformation,
-			KeyStage5PerformanceAdditionalInformation = detailsToUpdate.KeyStage5PerformanceAdditionalInformation ?? existingProject.Details.KeyStage5PerformanceAdditionalInformation
+			KeyStage5PerformanceAdditionalInformation = detailsToUpdate.KeyStage5PerformanceAdditionalInformation ?? existingProject.Details.KeyStage5PerformanceAdditionalInformation,
+			
+			// assigned user
+			AssignedUser = detailsToUpdate.AssignedUser != null 
+				? MapUser(detailsToUpdate.AssignedUser!.Id, detailsToUpdate.AssignedUser.FullName, detailsToUpdate.AssignedUser.EmailAddress)
```
[the remark this answers is not in the document]
Comment (jordan-beard) [reply Dfe.Academies.Academisation.Service/Mappers/Legacy/ProjectAggregate/LegacyProjectDetailsMapper.cs:115]: It's to get rid of the warning in this project.
Its telling the compiler that this Id will always have a value.
Examples of those warnings can be found here - 
image

[the remark this answers is not in the document]
Comment (jordan-beard) [reply Dfe.Academies.Academisation.Service/Mappers/Legacy/ProjectAggregate/LegacyProjectDetailsMapper.cs:115]: It's to get rid of the warning in this project.
Its telling the compiler that this Id will always have a value.
Examples of those warnings can be found here - 
image

**Review by dneed-nimble — APPROVED** [review]

**Pull request merged**